spot-contracts/contracts/FeePolicy.sol

```diff
@@ -0,0 +1,260 @@
+    uint256 public constant SIGMOID_BOUND = ONE / 100; // 0.01 or 1%
+
+    /// @notice Reference to the perpetual token.
+    IPerpetualTranche public perp;
+
+    /// @notice Reference to the rollover vault.
+    IVault public vault;
+
+    /// @notice The target subscription ratio i.e) the normalization factor.
+    uint256 public targetSubscriptionRatio;
+
+    /// @notice The percentage fee charged on minting perp tokens.
+    uint256 public perpMintFeePerc;
+
+    /// @notice The percentage fee charged on burning perp tokens.
+    uint256 public perpBurnFeePerc;
+
+    /// @notice The percentage fee charged on minting vault notes.
+    uint256 public vaultMintFeePerc;
+
+    /// @notice The percentage fee charged on burning vault notes.
+    uint256 public vaultBurnFeePerc;
+
+    /// @notice The fixed amount vault fee charged during each deployment.
+    /// @dev Denominated in the underlying collateral asset and
+    ///      Paid by the vault note holders to the system owner.
+    uint256 public vaultDeploymentFee;
+
+    struct RolloverFeeSigmoidParams {
+        /// @notice Lower asymptote
+        int256 lower;
+        /// @notice Upper asymptote
+        int256 upper;
+        /// @notice sigmoid slope
+        int256 growth;
+    }
+
+    /// @notice Parameters which control the asymptotes and the slope of the perp token's rollover fee.
+    RolloverFeeSigmoidParams public perpRolloverFee;
+
+    /// @dev The current subscription state the vault system relative to the perp supply.
+    struct SubscriptionState {
+        /// @notice The current tvl of perp.
+        uint256 currentPerpTVL;
+        /// @notice The current tvl of the rollover vault.
+        uint256 currentVaultTVL;
+        /// @notice The equilibrium tvl of the rollover vault.
+        uint256 equilibriumVaultTVL;
+        /// @notice Computed normalized subscription ratio.
+        uint256 normalizedSubscriptionRatio;
+    }
+
+    /// @notice Contract initializer.
+    /// @param perp_ Reference to perp.
+    /// @param vault_ Reference to the rollover vault.
+    function init(IPerpetualTranche perp_, IVault vault_) public initializer {
+        __Ownable_init();
+
+        perp = perp_;
+        vault = vault_;
+
+        perpMintFeePerc = 0;
+        perpBurnFeePerc = 0;
+
+        vaultMintFeePerc = 0;
+        vaultBurnFeePerc = 0;
+
+        perpRolloverFee.lower = -int256(ONE) / 30; // -0.033
+        perpRolloverFee.upper = int256(ONE) / 10; // 0.1
+        perpRolloverFee.growth = 5 * int256(ONE); // 5.0
+
+        targetSubscriptionRatio = (ONE * 133) / 100; // 1.33
+    }
+
+    /// @notice Updates the target subscription ratio.
+    /// @param targetSubscriptionRatio_ The new target subscription ratio as a fixed point number with {DECIMALS} places.
+    function updateTargetSubscriptionRatio(uint256 targetSubscriptionRatio_) external onlyOwner {
+        require(targetSubscriptionRatio_ > 0, "FeeStrategy: invalid subscription");
```

Comment:
consider upper/lower bounds to prevent decimal errors, e.g.

(ONE * 50) / 100 < targetSubscriptionRatio < ONE * 2